bin/theme-setup.js

```diff
@@ -131,21 +133,157 @@ const promptThemeDataShort = ({devUrl, noConfirm}) => {
+      { label: `${emoji.get('earth_africa')} Dev url`, variable: themeData.url },
+    ], noConfirm);
+  } while (confirmed !== 'y');
+
+  return themeData;
+};
+
+/**
+ * Prompts the user for all theme data (doesn't assume almost anything)
+ */
+const promptThemeData = ({ themeName, devUrl, noConfirm }) => {
+  let confirmed = 'n';
+  const themeData = {};
+
+  // -----------------------------
+  //  Prompt for project info
+  // -----------------------------
+
+  do {
+    if (!themeName) {
+      themeData.name = promptFor({
+        label: `${emoji.get('green_book')} Please enter your theme name (shown in WordPress admin):`,
+        prompt: 'Theme name: ',
+        error: 'Theme name field is required and cannot be empty.',
+        minLength: 2,
+      }).trim();
+    } else {
+      themeData.name = themeName;
+    }
+
+    // Build package name from theme name
+    themeData.package = themeData.name.toLowerCase().split(' ').join('_');
+
+    // Build prefix from theme name using one of 2 methods.
+    // 1. If theme name has 2 or mor more words, use first letters of each word
+    themeData.prefix = '';
+    const themeNameWords = themeData.name.split(' ');
+    if (themeNameWords && themeNameWords.length >= 2) {
+      for (const word of themeNameWords) {
+        themeData.prefix += word.charAt(0).toUpperCase();
+      }
+    }
+
+    // 2. If theme has only 1 word, use the first 3 letters of theme name
+    if (themeData.prefix.length < 2 && themeData.name.length > 2) {
+      themeData.prefix = (`${themeData.name.charAt(0)}${themeData.name.charAt(1)}${themeData.name.charAt(2)}`).toUpperCase();
+    }
+
+    themeData.env = `${themeData.prefix}_ENV`;
+    themeData.assetManifest = `${themeData.prefix}_ASSETS_MANIFEST`;
+
+    // Namespace
+    themeData.namespace = capCase(themeData.package);
+  
+    // Dev url
+    if (!devUrl) {
+      themeData.url = promptFor({
+        label: `${emoji.get('earth_africa')} Please enter a theme development url (for local development with browsersync - no protocol):`,
+        prompt: 'Dev url (e.g. dev.wordpress.com): ',
+        error: 'Dev url is required and cannot be empty.',
+      }).trim();
+    } else {
+      themeData.url = devUrl;
+    }
+
+    confirmed = summary([
+      {label: `${emoji.get('green_book')} Theme name`, variable: themeData.name},
+      {label: `${emoji.get('package')} Package`, variable: themeData.package},
+      {label: `${emoji.get('sun_behind_cloud')}  Namespace`, variable: themeData.namespace},
+      {label: `${emoji.get('bullettrain_front')} Theme prefix`, variable: themeData.prefix},
       {label: `${emoji.get('earth_africa')} Dev url`, variable: themeData.url},
     ], noConfirm);
   } while (confirmed !== 'y');
 
   return themeData;
 };
 
-const replaceThemeData = async(themeData) => {
+
+const replaceThemeData = async (themeData, replaceAll = false) => {
+
+  // Name
+  if (replaceAll) {
+    if (themeData.name) {
+      await replace({
```

Comment:
this pattern repeats three times, can it be extracted out and reused?

Reply:
That code is actually unused in this context (when using the **local** install script), so I just removed it. 👍